src/com/google/javascript/jscomp/deps/ModuleLoader.java

```diff
@@ -103,7 +105,21 @@ public ModuleLoader(
       case NODE:
         this.moduleResolver =
             new NodeModuleResolver(
-                this.modulePaths, this.moduleRootPaths, packageJsonMainEntries, this.errorHandler);
+                this.modulePaths, this.moduleRootPaths, lookupMap, this.errorHandler);
+        break;
+      case WEBPACK:
```

Comment:
I'm starting to think these constructors are complicated enough that they deserve to move to factory methods instead, but I'm not sure if that change makes sense to include now.

There aren't any callers of these constructors outside of Closure Compiler, FYI.

Reply:
> There aren't any callers of these constructors outside of Closure Compiler, FYI.

Well that gives me a little hope. I cringe every time I have to make a non-trivial change to the deps package.

But can we handle that as a separate PR?

Reply:
SGTM